jina/types/document/__init__.py

```diff
@@ -450,8 +453,40 @@ def get_attrs(self, *args) -> Dict[str, Any]:
         .. seealso::
             :meth:`update` for bulk set/update attributes
 
-        """
-        return {k: getattr(self, k) for k in args if hasattr(self, k)}
+        .. note::
+            Arguments not contained inside the `DocumentProto` definition, will be automatically extracted from `tags`.
+            All those arguments not in `DocumentProto` or not found in `tags` of the `Document`. will be extracted using `dunder_get`,
+            thus allowing to extract from `tags` or other `structures`.
+
+            .. highlight:: python
+            .. code-block:: python
+
+                d = Document({'id': '123', 'hello': 'world', 'tags': {'good': 'bye'}})
+                res = d.get_attrs(*['id', 'hello', 'tags__good'])
+
+                assert res['id'] == '123'
+                assert res['hello'] == 'world'
+                assert res['good'] == 'bye'
```

Comment:
I would prefer `assert res['tags_good'] == 'bye'`, but I guess having it as `good` is exactly the point, right? But what should happen if I do
```
 res = d.get_attrs(*['id', 'hello', 'tags__id'])
```
?

Reply:
yes, `tags__id` is a big problem. I think the nice way of assuming things from `tags` has the good thing that Executor developer does not need to know about `tags` and can keep a more "natural" language.

For instance, https://github.com/jina-ai/jina-hub/pull/3670 in this hub executor I propose, one needs to extract `features`. Should I know that they are stored in `tags`? Or is it better to allow the `Ranker` to talk its own language?

I see benefits and problems in every approach.

In the current approach there is a problem with `id` and `tags__id` because one would overwrite the other. It could be solved in a little hacky way fhough

Reply:
Handled with new logic